### PR TITLE
Don’t fail rename if there are multiple index entries for the same USR

### PR DESCRIPTION
**Pull request by ahoppen**
The test case that I’m adding right now is that we have a lingering index entry from a file that has been moved on disk. While the proper fix here would be to ignore such an entry from the index, we should also be more resilient and not fail rename altogether just because there are multiple entries for a USR – the chances are very high that they all translate to the same Swift and Objective-C name.

rdar://123937371

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Linux

File: Sources/SourceKitLSP/Rename.swift [hunk 1]
```diff
@@ -554,12 +554,14 @@ extension SourceKitServer {
...
+    if definitions.count > 1 {
+      logger.log("Multiple definitions for \(usr) found")
+    }
+    // There might be multiple definitions of the same symbol eg. in different `#if` branches. In this case pick any of
+    // them because with very high likelihood they all translate to the same clang and Swift name. Sort the entries to
+    // ensure that we deterministically pick the same entry every time.
+    guard let definitionSymbol = definitions.sorted().first else {
+      logger.error("no definitions for \(usr) found")
       return nil
     }
```
Comment (bnbarham) [inline Sources/SourceKitLSP/Rename.swift:566]: Could try the next if the picked definition isn't able to be translated?

**Review by bnbarham — APPROVED** [review]

File: Sources/SourceKitLSP/Rename.swift (hunk 1 above)
Comment (ahoppen) [reply Sources/SourceKitLSP/Rename.swift]: Good idea 👍🏽

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**